File: test_project/gen_code_java/bag.java
package demo;


public class Bag {
    
    
    public final int Id;
    
    
    public final String Name;
    
    
    public final java.util.List<Integer> PriceList;
    
    
    public final java.util.Map<String, Integer> PropMap;
    

    public Bag(int id, String name, java.util.List<Integer> pricelist, java.util.Map<String, Integer> propmap) {
        
        
        this.Id = id;
        
        this.Name = name;
        
        this.PriceList = pricelist;
        
        this.PropMap = propmap;
        
    }
}

File: test_project/gen_code_lua/bag.lua
-- Lua 侧通常通过 require table 来使用，Bean 通常对应的是 table
-- Bag 
local Bag = {}




function Bag.New(id, name, pricelist, propmap)
    local obj = {}
    
    
    obj.Id = id 
    
    obj.Name = name 
    
    obj.PriceList = pricelist 
    
    obj.PropMap = propmap 
    
    return obj
end

return Bag